core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1039,7 +1039,7 @@ object Chunk {
       else if (isFloat(c)) FloatClass.asInstanceOf[ClassTag[A]]
       else if (isDouble(c)) DoubleClass.asInstanceOf[ClassTag[A]]
       else if (isChar(c)) CharClass.asInstanceOf[ClassTag[A]]
-      else classTag[java.lang.Object].asInstanceOf[ClassTag[A]]
+      else classTag[AnyRef].asInstanceOf[ClassTag[A]] // TODO: Find a better way
```

Comment:
This was the problem with the `ArrayStoreException`?

Reply:
Yes. And I dug in deep, it literally has no stack trace attached, probably because it's generated from Scala code.

It happens when we have an array like `Array(Dog, Animal, Cat)` and detect the type as `Dog` and so specialize to `[LDog` but then putting `Animal` or `Cat` in there throws `ArrayStoreException`. So we either need to be not lazy or just use `AnyRef`. There's no reason I can tell for specializing the array, may as well just always do `AnyRef` arrays. But the types are a lie so we should encode that more nicely.